src/main.rs

```diff
@@ -295,7 +296,8 @@ fn main() {
     if !base_dir.exists() {
         create_dir_all(base_dir.clone()).unwrap();
     }
-    let url: &str = &config.listen_on.clone();
+    let url: &str = &std::env::var("PUBLISH_URL").unwrap_or(config.publish_url.clone());
```

Comment:
I would suggest to use the [`dotenv` crate](https://crates.io/crates/dotenv) here instead.